test/unit/sql/insert.test.js

```diff
@@ -93,9 +93,9 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
           bind: {
             ibmi: { sequelize_1: '2015-01-20 00:00:00.000' },
             sqlite: { sequelize_1: '2015-01-20 00:00:00.000 +00:00' },
-            db2: { sequelize_1: '2015-01-20 01:00:00' },
-            mysql: { sequelize_1: '2015-01-20 01:00:00' },
-            snowflake: { sequelize_1: '2015-01-20 01:00:00' },
+            db2: { sequelize_1: '2015-01-20 00:00:00.000' },
```

Comment:
DB2 doesn't have a way to set the timezone, so the timezone option must be ignored or it will end up offset

sqlite & ibmi also ignore the option, but they should still be tested against CET to ensure the option is ignored.

Reply:
Ok I've refactored test test a bit to make things clearer:

Only snowflake, mysql, mariadb and postgres support changing the global timezone. The others now throw (previously only sqlite and ibmi threw, now db2, mssql throw too).

Reply:
I've reworked the test to make it clearer, and made dialects that don't support this option throw

Reply:
Relevant commit to make it easier to review: https://github.com/sequelize/sequelize/pull/14505/commits/a71035e67c72ff62a70d46fe192f8ccd0157319d

Reply:
This change looks good. Thanks!